TestForceUpgradePendingPodsInOneStatefulSet is flaky

From: https://devops-ci.elastic.co/blue/organizations/jenkins/cloud-on-k8s-stack/detail/cloud-on-k8s-stack/151/pipeline

```
[2020-01-13T01:23:09.219Z] === Failed
[2020-01-13T01:23:09.219Z] === FAIL: test/e2e/es TestForceUpgradePendingPodsInOneStatefulSet/All_expected_Pods_should_eventually_be_ready (900.00s)
[2020-01-13T01:23:09.219Z] Retries (15m0s timeout): .........................................................................................................................................................................................................................................................................................................
[2020-01-13T01:23:09.219Z] {"level":"error","@timestamp":"2020-01-13T01:20:42.649Z","message":"stopping early","ver":"0.0.0-00000000","error":"test failure","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128\ngithub.com/elastic/cloud-on-k8s/test/e2e/test.StepList.RunSequential\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/step.go:42\ngithub.com/elastic/cloud-on-k8s/test/e2e/es.TestForceUpgradePendingPodsInOneStatefulSet\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/es/forced_upgrade_test.go:110\ntesting.tRunner\n\t/usr/local/go/src/testing/testing.go:909"}
[2020-01-13T01:23:09.220Z]     --- FAIL: TestForceUpgradePendingPodsInOneStatefulSet/All_expected_Pods_should_eventually_be_ready (900.00s)
[2020-01-13T01:23:09.220Z]         utils.go:83: 
[2020-01-13T01:23:09.220Z]             	Error Trace:	utils.go:83
[2020-01-13T01:23:09.220Z]             	Error:      	Received unexpected error:
[2020-01-13T01:23:09.220Z]             	            	pod force-upgrade-pending-sset-vflk-es-pending-0 is not Ready.
[2020-01-13T01:23:09.220Z]             	            	Status:{
[2020-01-13T01:23:09.220Z]             	            	    "phase": "Pending",
[2020-01-13T01:23:09.220Z]             	            	    "conditions": [
[2020-01-13T01:23:09.220Z]             	            	        {
[2020-01-13T01:23:09.220Z]             	            	            "type": "PodScheduled",
[2020-01-13T01:23:09.220Z]             	            	            "status": "False",
[2020-01-13T01:23:09.220Z]             	            	            "lastProbeTime": "2020-01-13T01:20:17Z",
[2020-01-13T01:23:09.220Z]             	            	            "lastTransitionTime": "2020-01-13T01:04:50Z",
[2020-01-13T01:23:09.220Z]             	            	            "reason": "Unschedulable",
[2020-01-13T01:23:09.220Z]             	            	            "message": "0/3 nodes are available: 3 node(s) didn't match node selector."
[2020-01-13T01:23:09.220Z]             	            	        }
[2020-01-13T01:23:09.220Z]             	            	    ],
[2020-01-13T01:23:09.220Z]             	            	    "qosClass": "Guaranteed"
[2020-01-13T01:23:09.220Z]             	            	}
[2020-01-13T01:23:09.220Z]             	Test:       	TestForceUpgradePendingPodsInOneStatefulSet/All_expected_Pods_should_eventually_be_ready
[2020-01-13T01:23:09.220Z] 
[2020-01-13T01:23:09.220Z] === FAIL: test/e2e/es TestForceUpgradePendingPodsInOneStatefulSet (954.79s)
```

It seems that the "pending" `StatefulSet` has not been updated by ECK during the test: 

```
            "apiVersion": "apps/v1",
            "kind": "StatefulSet",
            "metadata": {
                "creationTimestamp": "2020-01-13T01:04:49Z",
                "generation": 1,
                "labels": {
                    "common.k8s.elastic.co/template-hash": "407806796",
                    "common.k8s.elastic.co/type": "elasticsearch",
                    "elasticsearch.k8s.elastic.co/cluster-name": "force-upgrade-pending-sset-vflk",
                    "elasticsearch.k8s.elastic.co/statefulset-name": "force-upgrade-pending-sset-vflk-es-pending"
                },
...
                        "nodeSelector": {
                            "cannot": "be-scheduled"
                        },
...
```

--- a/pkg/controller/elasticsearch/bootstrap/bootstrap.go
+++ b/pkg/controller/elasticsearch/bootstrap/bootstrap.go
@@ -41,7 +41,19 @@ func ReconcileClusterUUID(k8sClient k8s.Client, cluster *esv1.Elasticsearch, esC
 	}
 	clusterUUID, err := getClusterUUID(esClient)
 	if err != nil {
-		return false, err
+		// There was an error while retrieving the UUID of the Elasticsearch cluster.
+		// For example, it could be the case with ES 6.x if the cluster does not have a master yet, in this case an
+		// API call to get the cluster UUID returns a 503 error.
+		// However we don't want to stop the reconciliation loop here because it could prevent the user to apply
+		// an update to the cluster spec to fix a problem.
+		// Therefore we just log the error and notify the driver that the reconciliation should be eventually re-queued.
+		log.Info(
+			"Recoverable error while retrieving Elasticsearch cluster UUID",
+			"namespace", cluster.Namespace,
+			"es_name", cluster.Name,
+			"error", err,
+		)
+		return true, nil
 	}
 	if !isUUIDValid(clusterUUID) {
 		// retry later

--- a/pkg/controller/elasticsearch/bootstrap/bootstrap_test.go
+++ b/pkg/controller/elasticsearch/bootstrap/bootstrap_test.go
@@ -101,8 +101,8 @@ func TestReconcileClusterUUID1(t *testing.T) {
 				esReachable: true,
 				esClient:    &fakeESClient{uuid: "", err: errors.New("error")},
 			},
-			wantRequeue:    false,
-			wantErr:        true,
+			wantRequeue:    true,
+			wantErr:        false,
 			wantAnnotation: "",
 		},
 		{

